### PR TITLE
Build: initialize base trivy go plugin

### PR DESCRIPTION
**Pull request by malikovmm**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	cmdArgs := append(trivyCommand, "--format", "json", "--output", tempFilePath)
+	if err := exec.Command("trivy", cmdArgs...).Run(); err != nil {
+		log.Fatalf("failed to build report: %v", err)
+	}
+
+}
+func getScanType() string {
+	return os.Args[1]
+}
```
Comment (afdesk) [inline main.go:27]: i have an idea.
we can try to parse the json as a basic report and, if the error appears we'll try to parse as a k8s report.

File: main.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+func main() {
+	trivyCommand := os.Args[1 : len(os.Args)-1]
+	tempFilePath := filepath.Join(os.TempDir(), tempFileName)
+	defer removeFile(tempFilePath)
+
+	cmdArgs := append(trivyCommand, "--format", "json", "--output", tempFilePath)
+	if err := exec.Command("trivy", cmdArgs...).Run(); err != nil {
+		log.Fatalf("failed to build report: %v", err)
+	}
```
Comment (afdesk) [inline main.go:22]: also, now we can check a case if `--output` file is already set, and just use it.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"path/filepath"
+)
+
+var (
+	tempFileName = "trivy-go-plugin-temp.json"
+)
+
+func main() {
+	trivyCommand := os.Args[1 : len(os.Args)-1]
+	tempFilePath := filepath.Join(os.TempDir(), tempFileName)
```
Comment (afdesk) [inline main.go:16]: now we can add the custom flag, eg. `--plugin-output` for the result.
it must be a required option.

to Trivy we shouldn't pass this flag

File: plugin.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+name: "trivy-go-plugin"
+repository: github.com/afdesk/trivy-go-plugin
+version: "0.0.1"
+usage: trivy go plugin
+description: |-
+  A Trivy plugin
+platforms:
+  -
+    uri: https://github.com/afdesk/trivy-go-plugin/releases/download/v0.0.1/trivy-go-plugin_0.0.1_Linux-amd64.tar.gz
+    bin: ./trivy-go-plugin
```
Comment (afdesk) [inline plugin.yaml:10]: we should add several platforms here: mac, windows, arm, etc

File: goreleaser.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+builds:
+  -
+    main: main.go
+    binary: trivy-go-plugin
+    ldflags:
+      - -s -w
+      - "-extldflags '-static'"
+      - -X main.version={{.Version}}
+    env:
+      - CGO_ENABLED=0
```
Comment (afdesk) [inline goreleaser.yml:11]: do we really need these options here?

File: goreleaser.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      - linux
+    goarch:
+      - amd64
+
+archives:
+  -
+    format: tar.gz
+    name_template: "{{.ProjectName}}_{{.Version}}_{{.Os}}-{{.Arch}}"
+    replacements:
+      linux: Linux
```
Comment (afdesk) [inline goreleaser.yml:22]: archive replacements is deprecated now
https://goreleaser.com/deprecations/#archivesreplacements

File: goreleaser.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    files:
+      - README.md
+      - LICENSE
+      - ./*.tpl
```
Comment (afdesk) [inline goreleaser.yml:26]: do we need it?

**Review by afdesk — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.19.0'
+      - run: git tag v${{ needs.check-version.outputs.next_version }}
+      - name: GoReleaser
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          version: latest
+          args: release -f=goreleaser.yml --rm-dist --timeout 60m
```
Comment (afdesk) [inline .github/workflows/build.yml:46]: --rm-dist is deprecated too, right?

**Review by afdesk — CHANGES_REQUESTED** [review]

File: main.go [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+func removeFile(file string) {
+	if err := os.Remove(file); err != nil {
+		log.Fatalf("failed to remove file %v", err)
+	}
+}
+func closeFile(file *os.File) {
+	if err := file.Close(); err != nil {
+		log.Fatalf("failed to remove file %v", err)
+	}
+}
```
Comment (afdesk) [inline main.go:41]: let's create a new function eg. `saveResult`. something like it

```suggestion
func saveResult(filename string, result []byte) error {
  f, err := os.CreateFile(filename)
  if err != nil {
     return err
  }
  
   if err := file.Close(); err != nil {
	return err
   }
   if err := os.Remove(file); err != nil {
	log.Printf("failed to remove file %v", err)
   }
   return nil
}

}
```

**Review by afdesk — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (malikovmm) [reply main.go:27]: there will be no error, just empty values will be written if using encoding/json

**Pull request merged**